### ISSUE TITLE
Add a colorbar to plot_connectome

### ISSUE THREAD

**Issue by GaelVaroquaux** [nilearn/nilearn#807]
There is no easy to have a colorbar in plot_connectome. We need to add this, controlled by a 'colorbar' argument.

**Comment by AlexandreAbraham** [comment]
For what it's worth, I have some code to create colorbar out of the blue:

```python
from matplotlib import pyplot as plt
from matplotlib import cm
import matplotlib as mpl
import numpy as np


def _axis_bbox_to_figure(fig, transform, bbox):
    # First, handle negative value for width and height
    bbox = list(bbox)
    if bbox[2] < 0:
        bbox[0] = bbox[0] + bbox[2]
        bbox[2] = (- bbox[2])
    if bbox[3] < 0:
        bbox[1] = bbox[1] + bbox[3]
        bbox[3] = (- bbox[3])
    inv_fig_t = fig.transFigure.inverted().transform
    ax_t = transform.transform
    tx, ty = inv_fig_t(ax_t(bbox[:2]))
    tw, th = inv_fig_t(ax_t(bbox[2:]) - ax_t([0, 0]))
    return tx, ty, tw, th


def plot_colorbar(cmap, ticks, orientation='vertical',
                  ticklabels=None,
                  continuous=True, bbox=None, ax=None, fig=None,
                  transform=None):
    if ax is None:
        if fig is None:
            fig = plt.gcf()
        # If there is a transform, handle it manually since matlplotlib is
        # shit
        if transform is not None:
            bbox = _axis_bbox_to_figure(fig, transform, bbox)
        ax = fig.add_axes(bbox)

    ticks = sorted(ticks)
    if continuous:
        norm = mpl.colors.Normalize(vmin=ticks[0], vmax=ticks[-1])
        ticks_ = None
    else:
        # Normalize the values
        ticks_ = np.linspace(0., 1., len(ticks) + 1)
        cmap = cm.get_cmap(cmap)
        cmap_ = mpl.colors.ListedColormap([cmap(t) for t in ticks_])
        norm = mpl.colors.BoundaryNorm(ticks_, cmap_.N)
        cmap = cmap_
        # Now get coordinates of real ticks
        ticks_ = [(ticks_[i] + ticks_[i + 1]) / 2.
                  for i in range(len(ticks))]

    cb = mpl.colorbar.ColorbarBase(ax, cmap=cmap,
                                   norm=norm,
                                   ticks=ticks_,
                                   orientation=orientation)

    if not continuous:
        # Labels
        if ticklabels is None:
            ticklabels = [str(t) for t in ticks]
        if orientation == 'vertical':
            cb.ax.set_yticklabels(ticklabels)
        elif orientation == 'horizontal':
            cb.ax.set_xticklabels(ticklabels)
    return cb
```

**Comment by KamalakerDadi** [comment]
@GaelVaroquaux Can you be more explicit about no easy ? Will ```_colobar_show``` in ```displays.py``` be not useful here ?

**Comment by lesteve** [comment]
> @GaelVaroquaux Can you be more explicit about no easy ? Will _colobar_show in displays.py be not useful here ?

I remember playing with this a while ago and I vaguely remember that it wasn't that simple to reuse our colorbar code. From my notes:

the colorbar stuff is done in add_overlay
so that passing cmap=edge_cmap and colorbar=True in plot_glass_brain
doesn't do anything ...

Something I hacked together in img_plotting.py (not even sure that was working TBH)
```python
    display = plot_glass_brain(None,
                               display_mode=display_mode,
                               figure=figure, axes=axes, title=title,
                               annotate=annotate,
                               black_bg=black_bg)

    class PseudoImage(object):
        def __init__(self, norm, cmap):
            self.norm = norm
            self.cmap = cmap

    from matplotlib import colors
    norm = colors.Normalize(vmin=-1, vmax=1)
    display._colorbar_show(PseudoImage(norm, edge_cmap), None)

    display.add_graph(adjacency_matrix, node_coords,
                      node_color=node_color, node_size=node_size,
                      edge_cmap=edge_cmap, edge_threshold=edge_threshold,
                      edge_kwargs=edge_kwargs, node_kwargs=node_kwargs)

```

Also I am not sure whether we want to make the colorbar depend on the values of the adjacency matrix.

**Comment by KamalakerDadi** [comment]
This issue can be closed since it is addressed in PR #1002
